platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/IssDetector.java

```diff
@@ -396,151 +62,22 @@ public void signalEndOfPreconsensusReplay(@Nullable final Object ignored) {
-                    "status is undecided, but method reported a decision, round = " + round);
-            case LACK_OF_DATA -> throw new IllegalStateException(
-                    "a decision that we lack data should only be possible once time runs out, round = " + round);
-            default -> throw new IllegalStateException(
-                    "unhandled case " + roundValidator.getStatus() + ", round = " + round);
-        };
-    }
-
-    /**
-     * This node doesn't agree with the consensus hash.
-     *
-     * @param roundHashValidator the validator responsible for validating the round with a self ISS
-     */
-    private void handleSelfIss(@NonNull final RoundHashValidator roundHashValidator) {
-        final long round = roundHashValidator.getRound();
-        final Hash selfHash = roundHashValidator.getSelfStateHash();
-        final Hash consensusHash = roundHashValidator.getConsensusHash();
-
-        final long skipCount = selfIssRateLimiter.getDeniedRequests();
-        if (selfIssRateLimiter.requestAndTrigger()) {
-
-            final StringBuilder sb = new StringBuilder();
-            sb.append("Invalid State Signature (ISS): this node has the wrong hash for round ")
-                    .append(round)
-                    .append(".\n");
-
-            roundHashValidator.getHashFinder().writePartitionData(sb);
-            writeSkippedLogCount(sb, skipCount);
-
-            logger.fatal(
-                    EXCEPTION.getMarker(),
-                    new IssPayload(sb.toString(), round, selfHash.toMnemonic(), consensusHash.toMnemonic(), false));
-        }
-    }
-
-    /**
-     * There has been a catastrophic ISS or a catastrophic lack of data.
-     *
-     * @param roundHashValidator information about the round, including the signatures that were gathered
-     */
-    private void handleCatastrophic(@NonNull final RoundHashValidator roundHashValidator) {
-
-        final long round = roundHashValidator.getRound();
-        final ConsensusHashFinder hashFinder = roundHashValidator.getHashFinder();
-        final Hash selfHash = roundHashValidator.getSelfStateHash();
-
-        final long skipCount = catastrophicIssRateLimiter.getDeniedRequests();
-        if (catastrophicIssRateLimiter.requestAndTrigger()) {
-
-            final StringBuilder sb = new StringBuilder();
-            sb.append("Catastrophic Invalid State Signature (ISS)\n");
-            sb.append("Due to divergence in state hash between many network members, "
-                    + "this network is incapable of continued operation without human intervention.\n");
-
-            hashFinder.writePartitionData(sb);
-            writeSkippedLogCount(sb, skipCount);
-
-            logger.fatal(EXCEPTION.getMarker(), new IssPayload(sb.toString(), round, selfHash.toMnemonic(), "", true));
-        }
-    }
+    @Nullable
+    List<IssNotification> overridingState(@NonNull ReservedSignedState state);
```

Comment:
Do these interface methods need annotation labels for the diagram?

Reply:
Currently not mandatory.

Reply:
Discussed this on slack, we agreed that this is ok to merge as is